Deleting processes results in exception

If you delete a process then the following error message will appear / get logged:

```
[DEBUG] 2021-08-13 13:56:59,332 [http-nio-127.0.0.1-8080-exec-21] org.kitodo.production.services.data.base.SearchService - org.kitodo.data.database.exceptions.DAOException: javax.persistence.EntityNotFoundException: deleted object would be re-saved by cascade (remove deleted object from associations): [org.kitodo.data.database.beans.Process#465926]
[ERROR] 2021-08-13 13:56:59,332 [http-nio-127.0.0.1-8080-exec-21] org.kitodo.production.services.command.KitodoScriptService - errorDeleting
org.kitodo.data.exceptions.DataException: org.kitodo.data.database.exceptions.DAOException: javax.persistence.EntityNotFoundException: deleted object would be re-saved by cascade (remove deleted object from associations): [org.kitodo.data.database.beans.Process#465926]
        at org.kitodo.production.services.data.base.SearchService.remove(SearchService.java:364) ~[classes/:3.4.0-SNAPSHOT]
        at org.kitodo.production.services.command.KitodoScriptService.deleteProcess(KitodoScriptService.java:352) [classes/:3.4.0-SNAPSHOT]
        at org.kitodo.production.services.command.KitodoScriptService.executeOtherScript(KitodoScriptService.java:191) [classes/:3.4.0-SNAPSHOT]
        at org.kitodo.production.services.command.KitodoScriptService.executeScript(KitodoScriptService.java:166) [classes/:3.4.0-SNAPSHOT]
        at org.kitodo.production.services.command.KitodoScriptService.execute(KitodoScriptService.java:121) [classes/:3.4.0-SNAPSHOT]
        at org.kitodo.production.forms.ProcessForm.executeKitodoScriptForProcesses(ProcessForm.java:658) [classes/:3.4.0-SNAPSHOT]
        at org.kitodo.production.forms.ProcessForm.executeKitodoScriptSelection(ProcessForm.java:652) [classes/:3.4.0-SNAPSHOT]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_292]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_292]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_292]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_292]
        at org.apache.el.parser.AstValue.invoke(AstValue.java:247) [tomcat8-jasper-el-8.5.54.jar:8.5.54]
        at org.apache.el.MethodExpressionImpl.invoke(MethodExpressionImpl.java:267) [tomcat8-jasper-el-8.5.54.jar:8.5.54]
        at org.jboss.weld.util.el.ForwardingMethodExpression.invoke(ForwardingMethodExpression.java:40) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
        at org.jboss.weld.el.WeldMethodExpression.invoke(WeldMethodExpression.java:50) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
        at org.jboss.weld.util.el.ForwardingMethodExpression.invoke(ForwardingMethodExpression.java:40) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
        at org.jboss.weld.el.WeldMethodExpression.invoke(WeldMethodExpression.java:50) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
        at org.apache.myfaces.view.facelets.el.ContextAwareTagMethodExpression.invoke(ContextAwareTagMethodExpression.java:96) [myfaces-impl-2.3.4.jar:2.3.4]
        at org.apache.myfaces.application.ActionListenerImpl.processAction(ActionListenerImpl.java:74) [myfaces-impl-2.3.4.jar:2.3.4]
        at org.springframework.faces.webflow.FlowActionListener.processAction(FlowActionListener.java:71) [spring-faces-2.5.1.RELEASE.jar:2.5.1.RELEASE]
        at org.springframework.faces.model.SelectionTrackingActionListener.processAction(SelectionTrackingActionListener.java:64) [spring-faces-2.5.1.RELEASE.jar:2.5.1.RELEASE]
        at javax.faces.component.UICommand.broadcast(UICommand.java:120) [myfaces-api-2.3.4.jar:2.3.4]
        at javax.faces.component.UIViewRoot._broadcastAll(UIViewRoot.java:1255) [myfaces-api-2.3.4.jar:2.3.4]
        at javax.faces.component.UIViewRoot.broadcastEvents(UIViewRoot.java:420) [myfaces-api-2.3.4.jar:2.3.4]
        at javax.faces.component.UIViewRoot._process(UIViewRoot.java:1741) [myfaces-api-2.3.4.jar:2.3.4]
        at javax.faces.component.UIViewRoot.processApplication(UIViewRoot.java:935) [myfaces-api-2.3.4.jar:2.3.4]
        at org.apache.myfaces.lifecycle.InvokeApplicationExecutor.execute(InvokeApplicationExecutor.java:42) [myfaces-impl-2.3.4.jar:2.3.4]
        at org.apache.myfaces.lifecycle.LifecycleImpl.executePhase(LifecycleImpl.java:195) [myfaces-impl-2.3.4.jar:2.3.4]
        at org.apache.myfaces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:142) [myfaces-impl-2.3.4.jar:2.3.4]
        at javax.faces.webapp.FacesServlet.service(FacesServlet.java:204) [myfaces-api-2.3.4.jar:2.3.4]
...
Caused by: org.kitodo.data.database.exceptions.DAOException: javax.persistence.EntityNotFoundException: deleted object would be re-saved by cascade (remove deleted object from associations): [org.kitodo.data.database.beans.Process#465926]
        at org.kitodo.data.database.persistence.BaseDAO.remove(BaseDAO.java:122) ~[kitodo-data-management-3.4.0-SNAPSHOT.jar:?]
        at org.kitodo.production.services.data.base.SearchDatabaseService.removeFromDatabase(SearchDatabaseService.java:74) ~[classes/:3.4.0-SNAPSHOT]
        at org.kitodo.production.services.data.base.SearchService.remove(SearchService.java:361) ~[classes/:3.4.0-SNAPSHOT]
        ... 92 more
Caused by: javax.persistence.EntityNotFoundException: deleted object would be re-saved by cascade (remove deleted object from associations): [org.kitodo.data.database.beans.Process#465926]
        at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:133) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
        at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:181) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
        at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:188) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
        at org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1364) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
        at org.hibernate.internal.SessionImpl.flush(SessionImpl.java:1347) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
        at org.kitodo.data.database.persistence.BaseDAO.remove(BaseDAO.java:118) ~[kitodo-data-management-3.4.0-SNAPSHOT.jar:?]
        at org.kitodo.production.services.data.base.SearchDatabaseService.removeFromDatabase(SearchDatabaseService.java:74) ~[classes/:3.4.0-SNAPSHOT]
        at org.kitodo.production.services.data.base.SearchService.remove(SearchService.java:361) ~[classes/:3.4.0-SNAPSHOT]
        ... 92 more
```

Error appears independend if you delete by the trash symbol in the process list or by calling KitodoScript "deleteprocess".

It looks like that the data of the process remains in process and task table - but it could be even in more tables.

--- a/Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java
+++ b/Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java
@@ -349,7 +349,7 @@ public class KitodoScriptService {
             } else {
                 try {
                     deleteMetadataDirectory(process);
-                    ServiceManager.getProcessService().remove(process);
+                    ServiceManager.getProcessService().deleteProcess(process);
                     Helper.setMessage("Process " + title + " deleted.");
                 } catch (DataException | IOException e) {
                     Helper.setErrorMessage("errorDeleting",

--- a/Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java
+++ b/Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java
@@ -348,7 +348,6 @@ public class KitodoScriptService {
                 }
             } else {
                 try {
-                    deleteMetadataDirectory(process);
                     ServiceManager.getProcessService().deleteProcess(process);
                     Helper.setMessage("Process " + title + " deleted.");
                 } catch (DataException | IOException e) {

